app/services/katello/pulp3/repository.rb

```diff
@@ -111,6 +115,17 @@ def get_distribution(href)
         nil
       end
 
+      def update_distributions
+        paths.values.select { |prefix, _path| prefix == 'http' }.each do |path|
+          dists = lookup_distributions(base_path: path.sub(/^\//, ''))
+          delete_distribution(dists.first._href) if dists.first
```

Comment:
Should this always delete the http distribution, or only if unprotected is false (and it exists)